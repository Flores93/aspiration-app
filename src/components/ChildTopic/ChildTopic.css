.child-topic {
  cursor: pointer;
  border: 1px solid rgba(70, 70, 201, 0.425);
  border-radius: 5px;
  margin-right: 5px;
  margin-bottom: 2px;
  padding: 2px;
}
.child-topic:hover {
  color: white;
  background-color: rgb(70, 70, 201);
  transition: 0.3s;
}
