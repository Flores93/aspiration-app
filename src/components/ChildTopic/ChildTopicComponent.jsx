import "./ChildTopic.css";

const ChildTopic = ({ node, changeTopic }) => {
  const {
    topic: { name, stargazerCount },
  } = node;
  return (
    <div className="child-topic" onClick={() => changeTopic(name)}>
      <b>{name}: </b>
      <span>
        {" "}
        {stargazerCount} <i className="fa fa-star mr-4 star-stargazers" />
      </span>
    </div>
  );
};

export default ChildTopic;
