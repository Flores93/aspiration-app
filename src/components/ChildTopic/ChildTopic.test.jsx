import { mount } from "enzyme";
import ChildTopic from "./ChildTopicComponent";

describe("Child Topic Component Test", () => {
  const changeTopicMock = jest.fn();

  const nodeMock = {
    topic: {
      name: "Pizza",
      stargazerCount: 9999,
    },
  };

  const wrapper = mount(
    <ChildTopic node={nodeMock} changeTopic={changeTopicMock} />
  );

  test("Should show the component correctly", () => {
    const validName = wrapper.find("b").text();
    const validStargazers = wrapper.find("span").text();

    expect(wrapper).toMatchSnapshot();
    expect(validName).toEqual(`${nodeMock.topic.name}: `);
    expect(validStargazers).toContain(nodeMock.topic.stargazerCount);
  });

  test("should call changeTopic function whit the name of the topic", () => {
    wrapper.find(".child-topic").simulate("click");

    expect(changeTopicMock).toHaveBeenCalledWith(nodeMock.topic.name);
  });
});
