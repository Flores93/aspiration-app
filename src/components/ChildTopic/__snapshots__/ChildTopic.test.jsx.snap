// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Child Topic Component Test Should show the component correctly 1`] = `
<div
  className="child-topic"
  onClick={[Function]}
>
  <b>
    Pizza
    : 
  </b>
  <span>
     
    9999
     
    <i
      className="fa fa-star mr-4 star-stargazers"
    />
  </span>
</div>
`;
